src/WebJobs.Script/Binding/Http/HttpRequestManager.cs

```diff
@@ -0,0 +1,151 @@
+
+                return await item.CompletionSource.Task;
+            }
+            else
+            {
+                // queue is not enabled, so just dispatch the request directly
+                return await processRequest(request, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// For a request that will be rejected due to load, max queue length
+        /// exceeded, etc. this method will be called, allowing the
+        /// status code, headers, etc. for the request to be configured.
+        /// </summary>
+        /// <param name="request">The request to reject.</param>
+        /// <returns>The response to return.</returns>
+        protected virtual HttpResponseMessage RejectRequest(HttpRequestMessage request)
+        {
+            return new HttpResponseMessage((HttpStatusCode)429);
+        }
+
+        /// <summary>
+        /// Returns a value indicating whether all incoming requests
+        /// should be rejected, for example due to host overload, etc.
+        /// </summary>
+        /// <returns>True if requests shoudl be rejected.</returns>
+        protected virtual bool RejectAllRequests()
+        {
+            return false;
+        }
+
+        private void InitializeRequestQueue()
+        {
+            var options = new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = Config.MaxDegreeOfParallelism
+            };
+
+            _requestQueue = new ActionBlock<HttpRequestItem>(async item =>
+            {
+                try
+                {
+                    var response = await item.ProcessRequest(item.Request, item.CancellationToken);
+                    item.CompletionSource.SetResult(response);
+                }
+                catch (Exception ex)
+                {
+                    item.CompletionSource.SetException(ex);
+                }
+
+            }, options);
+        }
+
+        private bool ThrottleExceeded()
+        {
+            if (_requestQueue == null)
+            {
+                return false;
+            }
+
+            bool queueLimitReached = Config.MaxQueueLength != DataflowBlockOptions.Unbounded && 
+                _requestQueue.InputCount >= Config.MaxQueueLength;
+
+            if (queueLimitReached)
+            {
+                TraceWriter.Info($"Http request queue limit of {Config.MaxQueueLength} has been exceeded.");
+            }
+
+            return queueLimitReached;
+        }
+
+        private class HttpRequestItem
+        {
+            /// <summary>
+            /// Gets or sets the request to process.
+            /// </summary>
+            public HttpRequestMessage Request { get; set; }
+
+            /// <summary>
+            /// Gets or sets the process methed for the request.
+            /// </summary>
+            public Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> ProcessRequest { get; set; }
```

Comment:
Given the fact that this is a delegate to the actual handler, I'd be more inclined to say this should be called `ProcessRequestHandler` or something along those lines.